api/src/database/run-ast.ts

```diff
@@ -50,7 +50,28 @@ export default async function runAST(
 		const results: { [collection: string]: null | Item | Item[] } = {};
 
 		for (const collection of ast.names) {
-			results[collection] = await run(collection, ast.children[collection], ast.query[collection]);
+			let hasMore = true;
+			let batchCount = 0;
+			results[collection] = [];
+
+			while (hasMore) {
+				const query = merge({}, ast.query[collection], {
+					limit: env.RELATIONAL_BATCH_SIZE,
+					offset: batchCount * env.RELATIONAL_BATCH_SIZE,
+					page: null,
+				});
+
+				const nestedItems = await run(collection, ast.children[collection], query);
+
+				if (Array.isArray(nestedItems)) results[collection]?.push(...nestedItems);
+				else results[collection]?.push(nestedItems);
```

Comment:
Fix resolves the linked issue. 👍 
Just a minor suggestion. I think the optional chaining should be replaced with `!` as the value is confirmed to be defined in line 55. 

```suggestion
				if (Array.isArray(nestedItems)) results[collection]!.push(...nestedItems);
				else results[collection]!.push(nestedItems);
```

Reply:
Thank you Ian 🙏